src/gt4py/next/ffront/foast_to_itir.py

```diff
@@ -427,18 +427,23 @@ def _visit_min_over(self, node: foast.Call, **kwargs: Any) -> itir.Expr:
         return self._make_reduction_expr(node, "minimum", init_expr, **kwargs)
 
     def _visit_type_constr(self, node: foast.Call, **kwargs: Any) -> itir.Expr:
-        if isinstance(node.args[0], foast.Constant):
-            node_kind = self.visit(node.type).kind.name.lower()
-            target_type = fbuiltins.BUILTINS[node_kind]
-            source_type = {**fbuiltins.BUILTINS, "string": str}[node.args[0].type.__str__().lower()]
-            if target_type is bool and source_type is not bool:
-                return im.promote_to_const_iterator(
-                    im.literal(str(bool(source_type(node.args[0].value))), "bool")
-                )
-            return im.promote_to_const_iterator(im.literal(str(node.args[0].value), node_kind))
-        raise FieldOperatorLoweringError(
-            f"Encountered a type cast, which is not supported: {node}."
-        )
+        el = node.args[0]
+        if isinstance(el, foast.Constant):
+            val, val_str = el.value, str(el.value)
+        elif isinstance(el, foast.UnaryOp):
+            assert hasattr(el.operand, "value")
```

Comment:
```suggestion
        elif isinstance(el, foast.UnaryOp) and isinstance(el.operand, foast.Constant):
```
Should run into the `else` clause giving a good error message in case the operand is not also a constant.